src/devices/acurite.c

```diff
@@ -713,36 +714,95 @@ static int acurite_txr_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+                if (decoder->verbose)
+                    fprintf(stderr, "%s: Acurite TXR sensor : bad channel Ch %s\n", __func__, channel_str);
+                continue;
+            }
+
             // Tower sensor ID is the last 14 bits of byte 0 and 1
             // CCII IIII | IIII IIII
             sensor_id = ((bb[0] & 0x3f) << 8) | bb[1];
-            //sensor_status = bb[2]; // TODO:, uses parity? & 0x07f
-            humidity = (bb[3] & 0x7f); // 1-99 %rH
-            // temperature encoding used by "tower" sensors 592txr
-            // 14 bits available after removing both parity bits.
-            // 11 bits needed for specified range -40 C to 70 C (-40 F - 158 F)
-            // range -100 C to 1538.4 C
-            int temp_raw = ((bb[4] & 0x7F) << 7) | (bb[5] & 0x7F);
-            tempc = temp_raw * 0.1 - 100;
+
             // Battery status is the 7th bit 0x40. 1 = normal, 0 = low
             battery_low = (bb[2] & 0x40) == 0;
 
-            /* clang-format off */
-            data = data_make(
-                    "model",                "",             DATA_STRING, "Acurite-Tower",
-                    "id",                   "",             DATA_INT,    sensor_id,
-                    "channel",              NULL,           DATA_STRING, channel_str,
-                    "battery_ok",           "Battery",      DATA_INT,    !battery_low,
-                    "temperature_C",        "Temperature",  DATA_FORMAT, "%.1f C", DATA_DOUBLE, tempc,
-                    "humidity",             "Humidity",     DATA_FORMAT, "%u %%", DATA_INT,    humidity,
-                    "mic",                  "Integrity",    DATA_STRING, "CHECKSUM",
-                    NULL);
-            /* clang-format on */
+            if (message_type == ACURITE_MSGTYPE_LEAK_DETECTOR) {
+                int is_wet = (bb[3] & 0x10) >> 4;
+
+                /* clang-format off */
+                data = data_make(
+                        "model",                "",             DATA_STRING, "Acurite-Leak",
+                        "id",                   "",             DATA_INT,    sensor_id,
+                        "channel",              NULL,           DATA_STRING, channel_str,
+                        "battery_ok",           "Battery",      DATA_INT,    !battery_low,
+                        "leak_detected",        "Leak",         DATA_INT,    is_wet,
+                        "mic",                  "Integrity",    DATA_STRING, "CHECKSUM",
+                        NULL);
+                /* clang-format on */
+            }
+            else if (message_type == ACURITE_MSGTYPE_TOWER_SENSOR) {
+
+                // Humidity is stored in byte 3
+                // The value is directly encoded as %rH
+                // The possible values here are 0-128, but real values are only 1-99 %rH
+                // pIII IIII
+                humidity = (bb[3] & 0x7f);
+                if (humidity < 1 || humidity > 99) {
```

Comment:
If this check allows the values 0 and 100, I'm fine with having this sanity check here until we can do something better.

Reply:
Changed to allow 0 and 100.

Reply:
But note that the value is from an unsigned source, the compiler will know that and look upon `< 0` in disgust (issue a compiler warning)

Reply:
I dealt with that by decoding it into a signed int. Since it's a 7 bit value, if anything <0 ever occurs, then something has gone __terribly__ wrong, and failing the decode is the least of our problems.

Reply:
At least Clang still knows where it came from and the possible range (a good thing). Checking for `< 0` will produce a warning.

Reply:
clang 12 seems quite happy to compile the refactored version of this code. No warnings.

Reply:
Ah ok, I remember seeing this on some other code but it's not detected here. Though my CppCheck plugin is not satisfied still.

Reply:
Interesting.... cpcheck has no complaints for me. It is mad about other pre-existing functions for which variable scope could be reduced, but that's it. Perhaps you have some additional checks turned on somewhere that I don't?

Reply:
I'm not aware of setting any custom options, hmm.
![cppchk](https://user-images.githubusercontent.com/12833/151187432-622f8bce-bd58-4423-bfb3-5deee674b108.png)

Reply:
Very odd. I'm running CppCheck via an extension in VSCode, and not getting any squigglies:
![image](https://user-images.githubusercontent.com/772977/151188371-55b13452-7c1c-44ec-904c-5b031f86f231.png)
Clearly we have something configured differently, I just don't know what/where.